Content.Shared/Ninja/Systems/SharedNinjaSuitSystem.cs

```diff
@@ -39,6 +43,11 @@ private void OnMapInit(EntityUid uid, NinjaSuitComponent component, MapInitEvent
         Dirty(uid, component);
     }
 
+    private void OnEntityUnpaused(Entity<NinjaSuitComponent> ent, ref EntityUnpausedEvent args)
+    {
+        ent.Comp.DisableCooldown += args.PausedTime;
```

Comment:
i rarely see anything using timespans for things do this is this good for any future things

Reply:
That's because a lot of things do not properly support pausing as it's something not that advertised on the docs AND it's easy to miss it when reviewing. But this is the correct way of handling these types of timespans (NOT the ones that have a set duration like the suit `DisableTime`). Take a look at this example:
```
ROUND TIME: 00:01:05

ROUND TIME AFTER PAUSING FOR 20 SECONDS: 00:01:25

SUIT COOLDOWN WITHOUT PAUSING HANDLING: ROUND TIME + 5 = 00:01:10

RESULT: Round unpauses and the cooldown is instantly dropped even tho no time should 
have passed.

SUIT COOLDOWN WITH PAUSING HANDLING: ROUND TIME + PAUSE TIME + 5 = 00:01:30

RESULT: Round unpuases and the cooldown continues as usual, as if no time had passed
```

Reply:
ok so just a mountain of old code needs to support it :trollface:

Reply:
basically